src/group_theory/nilpotent.lean

```diff
@@ -395,3 +397,16 @@ begin
     (le_trans ((map_eq_bot_iff _).mp _) hf1)⟩,
   exact eq_bot_iff.mpr (hn ▸ (lower_central_series.map f n)),
 end
+
+lemma derived_le_lower_central (n : ℕ) : derived_series G n ≤ lower_central_series G n :=
+by { induction n with i ih, { simp }, { apply general_commutator_mono ih, simp } }
+
+theorem is_nilpotent.is_solvable (h : is_nilpotent G): is_solvable G :=
+begin
+  rw nilpotent_iff_lower_central_series at *,
+  rcases h with ⟨n, hn⟩,
+  use n,
+  apply le_bot_iff.mp,
```

Comment:
Just so you know, you can use `refine` instrad of `use`.
```suggestion
  refine ⟨n, le_bot_iff.mp _⟩
```

Reply:
Is this a recommendation for here, or just a FYI? I (naively) think the `use` is helpful to spot the instantiation of the ∃. But it’s a matter of style, of course.

Reply:
Up to you, for for such short instantiations we usually don't bother using `use`.